### PR TITLE
Safe Get Key Error for Template Progress

### PR DESCRIPTION
**Pull request by oakeyc**
re-fix from PR #3737
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] The PR has modified HISTORY.rst with an appropriate description of the change (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [x] Each command and parameter has a meaningful description.
- [x] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

File: src/azure-cli-core/azure/cli/core/commands/__init__.py [hunk 1]
```diff
@@ -195,7 +195,7 @@ def _template_progress(self, correlation_id):  # pylint: disable=no-self-use
                         # don't want to show the timestamp
                         json_val = deploy_values.copy()
                         json_val.pop('timestamp', None)
-                        if deploy_values.get('status value', None) != 'Started':
+                        if deploy_values.get('status value', 'Started') != 'Started':
```
Comment (tjprescott) [inline src/azure-cli-core/azure/cli/core/commands/__init__.py]: Why would we assume a value of "Started" if the value contains no reference to status at all?

Comment (oakeyc) [inline src/azure-cli-core/azure/cli/core/commands/__init__.py]: You're right, that it's not very clear what it's trying to accomplish.

File: src/azure-cli-core/azure/cli/core/commands/__init__.py [hunk 2]
```diff
@@ -195,7 +195,8 @@ def _template_progress(self, correlation_id):  # pylint: disable=no-self-use
                         # don't want to show the timestamp
                         json_val = deploy_values.copy()
                         json_val.pop('timestamp', None)
-                        if deploy_values.get('status value', None) != 'Started':
+                        status_val = deploy_values.get('status value', None)
```
Comment (tjprescott) [inline src/azure-cli-core/azure/cli/core/commands/__init__.py]: So what output is produced when the deployment doesn't provide the "status value"?

Comment (oakeyc) [inline src/azure-cli-core/azure/cli/core/commands/__init__.py]: There would be none

**Pull request merged**